### PR TITLE
Fixes adjust config

### PR DESCRIPTION
**Pull request by AnturK**
Closes #30651
Changes config format because config depending on order is insane.
Also bit more meaningful name.

### CODE REVIEW

File: code/controllers/configuration.dm [hunk 1]
```diff
@@ -966,12 +968,13 @@ GLOBAL_PROTECT(config_dir)
 			M.maximum_players = max_pop[M.config_tag]
 		if(M.can_start())
 			var/final_weight = probabilities[M.config_tag]
-			var/recent_round = SSpersistence.saved_modes.Find(M.config_tag)
-			var/adjustment
-			while(recent_round)
-				adjustment += adjust[recent_round]
-				recent_round = SSpersistence.saved_modes.Find(M.config_tag,recent_round+1,0)
-			final_weight *= ((100-adjustment)/100)
+			if(SSpersistence.saved_modes.len == 3 && adjust.len == 3)
```
Comment (RemieRichards) [inline code/controllers/configuration.dm]: repeated_mode_adjust

**Comment by Robustin** [comment]
This fixes the runtime too?

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -710,9 +710,11 @@ GLOBAL_PROTECT(config_dir)
 							WRITE_FILE(GLOB.config_error_log, "Unknown game mode probability configuration definition: [prob_name].")
 					else
 						WRITE_FILE(GLOB.config_error_log, "Incorrect probability configuration definition: [prob_name]  [prob_value].")
-				if("adjust")
+				if("repeated_mode_adjust")
 					if(value)
```
Comment (Cyberboss) [inline code/controllers/configuration.dm]: should probably .Cut() the list before this, in case the config entry is duped

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/controllers/configuration.dm (hunk 2 above)
Comment (AnturK) [reply code/controllers/configuration.dm]: Sure.

**Comment by AnturK** [comment]
@Robustin Yes. You forgot to check if the config actually exists.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**